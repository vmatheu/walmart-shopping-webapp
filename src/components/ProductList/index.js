import React from 'react';
import { arrayOf, object } from 'prop-types';
import './style.css';

const ProductList = ({ products }) => (
  <div className="card-deck mb-3 text-center">
    {products.map(product => (
      <div key={product.id} className="ard mb-4 shadow-sm walmark-card">
        <div className="card-header">
          <h4 className="my-0 font-weight-normal">{product.description}</h4>
        </div>
        <div className="card-body">
          { product.priceHaveDiscount ?
            <div className="card-title pricing-card-title">
              <h3> ${product.price} <span className="walmark-discount">50%</span></h3>
              <h5 className="walmark-discount-price"> ${product.priceWithOutDiscount} </h5>
            </div>
          : <h3 className="card-title pricing-card-title">${product.price}</h3>}
          <img alt={`imagen del producto ${product.description}`} src={`http://${product.image}`} />
          <h6>{product.brand}</h6>
        </div>
      </div>
    ))}
  </div>
);

ProductList.propTypes = {
  products: arrayOf(object).isRequired,
};

export default ProductList;
