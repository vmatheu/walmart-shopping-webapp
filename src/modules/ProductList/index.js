import React from 'react';
import { arrayOf, object } from 'prop-types';

const ProductList = ({ products }) => (
  <div className="card-deck mb-3 text-center">
    {products.map(product => (
      <div key={product.id} className="ard mb-4 shadow-sm">
        <div className="card-header">
          <h4 className="my-0 font-weight-normal">{product.description}</h4>
        </div>

        <div className="card-body">
          { product.priceHaveDiscount ?
            <p className="card-title pricing-card-title">
              <h3> ${product.price} <span style={{ color: 'red' }}>50%</span></h3>
              <h5 style={{ 'text-decoration': 'line-through', color: 'grey' }}> ${product.priceWithOutDiscount} </h5>
            </p>
          : <h3 className="card-title pricing-card-title">${product.price}</h3>}
          <img alt={`imagen del producto ${product.description}`} src={`http://${product.image}`} />
          <h6>{product.brand}</h6>
        </div>
      </div>
    ))}
  </div>
);

ProductList.propTypes = {
  products: arrayOf(object).isRequired,
};

export default ProductList;
