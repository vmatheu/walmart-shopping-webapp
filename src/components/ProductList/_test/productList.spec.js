import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ProductList from '..';

const buildProduct = (id, brand, description, price, image) =>
  ({ id, brand, description, price, image });

describe('<ProductList/>', () => {
  describe('should be list product', () => {
    it('should have the same ui structure with one product', () => {
      const products = [
        buildProduct(1, 'coca cola', 'bebida', '$10', 'uriimage'),
      ];

      const wrapper = shallow(<ProductList products={products} />);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
    });

    it('should have the same ui structure with five product', () => {
      const products = [
        buildProduct(1, 'coca cola', 'bebida', '$10', 'uriimage'),
        buildProduct(2, 'pepsi', 'bebida', '$10', 'uriimage1'),
        buildProduct(3, 'fanta', 'bebida', '$10', 'uriimage2'),
        buildProduct(4, 'sprite', 'bebida', '$10', 'uriimage3'),
        buildProduct(5, 'bilz', 'bebida', '$10', 'uriimage4'),
      ];

      const wrapper = shallow(<ProductList products={products} />);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
    });
  });
});
