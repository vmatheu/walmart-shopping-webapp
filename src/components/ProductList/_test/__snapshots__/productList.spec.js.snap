// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductList/> should be list product should have the same ui structure with five product 1`] = `
<div
  className="card-deck mb-3 text-center"
>
  <div
    className="ard mb-4 shadow-sm walmark-card"
    key="1"
  >
    <div
      className="card-header"
    >
      <h4
        className="my-0 font-weight-normal"
      >
        bebida
      </h4>
    </div>
    <div
      className="card-body"
    >
      <h3
        className="card-title pricing-card-title"
      >
        $
        $10
      </h3>
      <img
        alt="imagen del producto bebida"
        src="http://uriimage"
      />
      <h6>
        coca cola
      </h6>
    </div>
  </div>
  <div
    className="ard mb-4 shadow-sm walmark-card"
    key="2"
  >
    <div
      className="card-header"
    >
      <h4
        className="my-0 font-weight-normal"
      >
        bebida
      </h4>
    </div>
    <div
      className="card-body"
    >
      <h3
        className="card-title pricing-card-title"
      >
        $
        $10
      </h3>
      <img
        alt="imagen del producto bebida"
        src="http://uriimage1"
      />
      <h6>
        pepsi
      </h6>
    </div>
  </div>
  <div
    className="ard mb-4 shadow-sm walmark-card"
    key="3"
  >
    <div
      className="card-header"
    >
      <h4
        className="my-0 font-weight-normal"
      >
        bebida
      </h4>
    </div>
    <div
      className="card-body"
    >
      <h3
        className="card-title pricing-card-title"
      >
        $
        $10
      </h3>
      <img
        alt="imagen del producto bebida"
        src="http://uriimage2"
      />
      <h6>
        fanta
      </h6>
    </div>
  </div>
  <div
    className="ard mb-4 shadow-sm walmark-card"
    key="4"
  >
    <div
      className="card-header"
    >
      <h4
        className="my-0 font-weight-normal"
      >
        bebida
      </h4>
    </div>
    <div
      className="card-body"
    >
      <h3
        className="card-title pricing-card-title"
      >
        $
        $10
      </h3>
      <img
        alt="imagen del producto bebida"
        src="http://uriimage3"
      />
      <h6>
        sprite
      </h6>
    </div>
  </div>
  <div
    className="ard mb-4 shadow-sm walmark-card"
    key="5"
  >
    <div
      className="card-header"
    >
      <h4
        className="my-0 font-weight-normal"
      >
        bebida
      </h4>
    </div>
    <div
      className="card-body"
    >
      <h3
        className="card-title pricing-card-title"
      >
        $
        $10
      </h3>
      <img
        alt="imagen del producto bebida"
        src="http://uriimage4"
      />
      <h6>
        bilz
      </h6>
    </div>
  </div>
</div>
`;

exports[`<ProductList/> should be list product should have the same ui structure with one product 1`] = `
<div
  className="card-deck mb-3 text-center"
>
  <div
    className="ard mb-4 shadow-sm walmark-card"
    key="1"
  >
    <div
      className="card-header"
    >
      <h4
        className="my-0 font-weight-normal"
      >
        bebida
      </h4>
    </div>
    <div
      className="card-body"
    >
      <h3
        className="card-title pricing-card-title"
      >
        $
        $10
      </h3>
      <img
        alt="imagen del producto bebida"
        src="http://uriimage"
      />
      <h6>
        coca cola
      </h6>
    </div>
  </div>
</div>
`;
